app/presenters/tree_builder_timelines.rb

```diff
@@ -0,0 +1,78 @@
+class TreeBuilderTimelines < TreeBuilder
+  has_kids_for Hash, [:x_get_tree_hash_kids]
+
+  def override(node, object, _pid, _options)
+    node[:expand] = object[:expand] if object[:expand].present?
+  end
+
+  def initialize(name, type, sandbox, build = true, menu:)
+    @menu = menu
+    super(name, type, sandbox, build)
+  end
+
+  private
+
+  def set_locals_for_render
+    super.merge!(
+      :id_prefix  => 'timelines_',
+      :onclick    => "miqOnClickTimelineSelection",
+      :click_url  => "/dashboard/show_timeline/",
+      :tree_state => true
+    )
+  end
+
+  def tree_init_options(_tree_name)
+    {
+      :full_ids => true,
+      :lazy     => false,
+      :add_root => false
+    }
+  end
+
+  def root_options
+    {}
+  end
+
+  def x_get_tree_roots(count_only, _options)
+    menus = @menu.map do |item|
+      {
+        :text        => t = item.first,
+        :id          => "r__#{t}",
+        :icon        => 'pficon pficon-folder-close',
+        :tip         => t,
+        :cfmeNoClick => true,
+        :expand      => true,
+        :subsections => item.last
+      }
+    end
+    count_only_or_objects(count_only, menus.compact)
+  end
+
+  def x_get_tree_hash_kids(parent, count_only)
+    subsections = parent[:subsections].map do |item|
+      if item.kind_of?(Array)
+        {
+          :text        => t = item.first,
```

Comment:
I would extract this `t = item.first` to before the hash.  It is not obvious this way as it is also used in other hash values below.